ml_git/relationship/local_entity_manager.py

```diff
@@ -0,0 +1,186 @@
+"""
+© Copyright 2021 HP Development Company, L.P.
+SPDX-License-Identifier: GPL-2.0-only
+"""
+
+import io
+from collections import namedtuple
+
+from git import Repo
+
+from ml_git import log
+from ml_git._metadata import MetadataManager
+from ml_git.config import config_load
+from ml_git.constants import SPEC_EXTENSION, FileType, EntityType
+from ml_git.ml_git_message import output_messages
+from ml_git.relationship.models.entity import Entity
+from ml_git.relationship.models.entity_version_relationships import EntityVersionRelationships
+from ml_git.relationship.models.linked_entity import LinkedEntity
+from ml_git.relationship.models.spec_version import SpecVersion
+from ml_git.relationship.utils import export_relationships_to_csv, export_relationships_to_dot
+from ml_git.utils import yaml_load_str, get_root_path
+
+
+class LocalEntityManager:
+    """Class that operate over local git repository to manage entity's operations"""
+
+    def __init__(self):
+        self._manager = None
+
+    def __init_manager(self, entity_type):
+        try:
+            get_root_path()
+            config = config_load()
+            if not config[entity_type]['git']:
+                log.warn(output_messages['ERROR_REPOSITORY_NOT_FOUND'])
+                return
+            self._manager = MetadataManager(config, repo_type=entity_type)
+            if not self._manager.check_exists():
+                self._manager.init()
+        except Exception as e:
+            log.error(e)
+
+    def get_entities(self):
+        """Get a list of entities found in config.yaml.
+
+        Returns:
+            list of class Entity.
+        """
+        entities = []
+        metadata_repository = namedtuple('Repository', ['private', 'full_name', 'ssh_url', 'html_url', 'owner'])
+        metadata_owner = namedtuple('Owner', ['email', 'name'])
+        for e_type in EntityType:
+            self.__init_manager(e_type.value)
+            if not self._manager:
+                continue
+            repository = metadata_repository(False, '', '', '', metadata_owner('', ''))
+            for obj in Repo(self._manager.path).head.commit.tree.traverse():
+                if SPEC_EXTENSION in obj.name:
+                    entity_spec = yaml_load_str(io.BytesIO(obj.data_stream.read()))
+                    entity = Entity(repository, entity_spec)
+                    if entity.type in e_type.value and entity not in entities:
+                        entities.append(entity)
```

Comment:
```suggestion
        for entity_type in EntityType:
            self.__init_manager(entity_type.value)
            if not self._manager:
                continue
            repository = metadata_repository(False, '', '', '', metadata_owner('', ''))
            for obj in Repo(self._manager.path).head.commit.tree.traverse():
                if SPEC_EXTENSION in obj.name:
                    entity_spec = yaml_load_str(io.BytesIO(obj.data_stream.read()))
                    entity = Entity(repository, entity_spec)
                    if entity.type in entity_type.value and entity not in entities:
                        entities.append(entity)
```

Is it not better use this way?

Reply:
Adjusted in cdb61294.